zerver/lib/narrow.py

```diff
@@ -676,6 +679,25 @@ def by_dm_including(
         )
         return query.where(maybe_negate(cond))
 
+    def by_mention(
+        self, query: Select, operand: str | int, maybe_negate: ConditionTransform
+    ) -> Select:
+        assert self.user_profile is not None
+
+        try:
+            if isinstance(operand, int):
+                user = get_user_by_id_in_realm_including_cross_realm(int(operand), self.realm)
+            else:
+                user = access_user_by_email(self.user_profile, operand, for_admin=False)
+        except JsonableError:
+            raise BadNarrowOperatorError("unknown user " + str(operand))
+
+        cond = (column("user_profile_id", Integer) == user.id) & (
+            column("flags", Integer).op("&")(literal(UserMessage.flags.mentioned.mask)) != 0
```

Comment:
We should add a comment here being very clear that this is intended to only cover direct personal mentions, so we don't check the other mention flags.